packages/sequelize/src/index.ts

```diff
@@ -0,0 +1,156 @@
+import type { Account as ApadterAccount } from "next-auth"
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterSession,
+  VerificationToken,
+} from "next-auth/adapters"
+import { Sequelize, Model, ModelCtor } from "sequelize"
+import {
+  accountSchema,
+  userSchema,
+  sessionSchema,
+  verificationTokenSchema,
+} from "./schema"
+
+// @see https://sequelize.org/master/manual/typescript.html
+interface AccountInstance
+  extends Model<ApadterAccount, Partial<ApadterAccount>>,
+    ApadterAccount {}
+interface UserInstance
+  extends Model<AdapterUser, Partial<AdapterUser>>,
+    AdapterUser {}
+interface SessionInstance
+  extends Model<AdapterSession, Partial<AdapterSession>>,
+    AdapterSession {}
+interface VerificationTokenInstance
+  extends Model<VerificationToken, Partial<VerificationToken>>,
+    VerificationToken {}
+
+export { accountSchema, userSchema, sessionSchema, verificationTokenSchema }
+
+export default function SequelizeAdapter(client: Sequelize): Adapter {
+  const tableOptions = { underscored: true, timestamps: false }
+
+  const Account =
+    (client.models.Account as ModelCtor<any>) ||
+    client.define<AccountInstance>("account", accountSchema, tableOptions)
+  const User =
+    (client.models.User as ModelCtor<any>) ||
+    client.define<UserInstance>("user", userSchema, tableOptions)
+  const Session =
+    (client.models.Session as ModelCtor<any>) ||
+    client.define<SessionInstance>("session", sessionSchema, tableOptions)
+  const VerificationToken =
+    (client.models.VerificationToken as ModelCtor<any>) ||
+    client.define<VerificationTokenInstance>(
+      "verificationToken",
+      verificationTokenSchema,
+      tableOptions
+    )
+
+  Account.belongsTo(User, { onDelete: "cascade" })
+  Session.belongsTo(User, { onDelete: "cascade" })
+
+  return {
+    async createUser(user) {
+      return await User.create(user)
+    },
+    async getUser(id) {
+      const userInstance = await User.findByPk(id)
+
+      return userInstance?.get({ plain: true }) ?? null
+    },
+    async getUserByEmail(email) {
+      const userInstance = await User.findOne({
+        where: { email },
+      })
+
+      return userInstance?.get({ plain: true }) ?? null
+    },
+    async getUserByAccount({ provider, providerAccountId }) {
+      const accountInstance = await Account.findOne({
+        where: { provider, providerAccountId },
+      })
+
+      if (!accountInstance) {
+        return null
+      }
+
+      const userInstance = await User.findByPk(accountInstance.userId)
+
+      return userInstance?.get({ plain: true }) ?? null
+    },
+    async updateUser(user) {
+      await User.update(user, { where: { id: user.id } })
+      const userInstance = await User.findByPk(user.id)
+
+      if (!userInstance) {
+        throw new Error("User does not exist")
+      }
+
+      return userInstance
```

Comment:
```suggestion
      return userInstance!
```

You could assume this never happens because we won't try updating a user from the core that does not exist.

(you will need to add `/* eslint-disable @typescript-eslint/no-non-null-assertion */` to the top of the file to be able to use the `!` TS non-null assertion)

Reply:
Nice, done in 3b2da67